tools/perf_analysis/test_broadcast_perf.py

```diff
@@ -0,0 +1,299 @@
+from matplotlib import pyplot as plt
+import json
+import os
+import sys
+import numpy as np
+import math
+from datetime import datetime
+
+#                 test_broadcast_perf.py
+#                 Author:  Brian Pickrell, AMD copyright 2022
+
+#
+# this script runs the test program gpu-driver while iterating over
+# ranges of
+# tensor lens; global work items; and local work items, and plots the results.
+# It saves output to a JSON File as well, so that it can be re-plotted by
+# other tools.
+
+# this script is simlar to test_pointwise_perf.py, except that its
+# Json arguments include "strides".  Setting one element in a stride to 0
+# cause a tensor to have one axis broadcast in gpu-driver.
+#
+# This script version is for comparing speeds of operators when broadcasting
+# tensor data.
+#
+# Engineers using this script should expect to modify it frequently in the course of testing.
```

Comment:
This file should also import `argparse` so that users can pass in modified values without editing the script each time.

Reply:
They'll end up editing it all the time anyway, but a placeholder arg parser has been added to both scripts.